### PR TITLE
Changes for international languages at virtual-alexa

### LINKED ISSUE #17: Japanese Support
Validate if multi language support actual code is all it's needed for Japanese to work correctly.

If not made the additions necessary for it.

### PR DESCRIPTION
**Pull request by jperata**
Removed utterances a-Z validation since that affects international words (as they have extra symbols as in ü), or completely different characters ( as in 人のプレ).

Added also a japanese test model and corresponding tests
  
Closes  #17

### CODE REVIEW

File: src/SampleUtterances.ts [hunk 1]
```diff
@@ -55,11 +55,10 @@ export class SampleUtterances {
     }
 
     public samplesForIntent(intent: string): SamplePhrase [] {
-        let samples: SamplePhrase[] = [];
-        if (intent in this.samples) {
-            samples = this.samples[intent];
+        if (!(intent in this.samples)) {
```
Comment (jkelvie) [inline src/SampleUtterances.ts]: Is this fixing a bug? I prefer to not do early returns. Do we have a philosophy on this?

File: src/SampleUtterances.ts [hunk 2]
```diff
@@ -160,8 +159,7 @@ export class SamplePhraseTest {
     private matchString: string;
 
     public constructor(public samplePhrase: SamplePhrase, private utterance: string) {
-        const cleanUtterance = utterance.replace(/[^0-9a-zA-Z ]/g, "");
```
Comment (jkelvie) [inline src/SampleUtterances.ts]: Maybe I left the unit tests out (which would be bad) but I think this was meant to deal with people putting in question marks and exclamation points and things like that. Will the code still work if someone adds special characters like those (which we want to ignore)?

Comment (jperata) [reply src/SampleUtterances.ts]: Makes sense, added the corresponding test and change the logic from allowing only a-z  and 0-9 characters to exclude symbols

File: src/SampleUtterances.ts (hunk 1 above)
Comment (jkelvie) [reply src/SampleUtterances.ts]: Okay, as long as you are applying this across our codebases, I am fine with this approach

File: test/UtteranceTest.ts [hunk 3]
```diff
@@ -203,5 +207,27 @@ describe("UtteranceTest", function() {
             assert.isTrue(utterance.matched());
             assert.equal(utterance.intent(), "NumberSlot");
         });
+
+        it("Matches with symbols in the phrase", () => {
+            const utterance = new Utterance(model, "good? #%.morning");
```
Comment (jkelvie) [inline test/UtteranceTest.ts]: The tests should also include an utterance with:
,
-
:
Those are very common punctuation marks

Comment (jperata) [reply test/UtteranceTest.ts]: added one more test for this one, were missing the character exclusions too

**Pull request merged**